source/common/http/conn_manager_impl.cc

```diff
@@ -857,10 +859,20 @@ void ConnectionManagerImpl::ActiveStream::decodeData(ActiveStreamDecoderFilter*
   if (!filter) {
     entry = decoder_filters_.begin();
   } else {
-    entry = std::next(filter->entry());
+    if ((*(filter->entry()))->stoppedAll()) {
+      // If filter's iteration_state_ has been stopped, its decodeData() has not be called. Call it
+      // now.
+      entry = filter->entry();
+    } else {
+      entry = std::next(filter->entry());
+      if (entry != decoder_filters_.end() && (*entry)->stoppedAll()) {
```

Comment:
Can you clarify in a comment how we get inside this if statement? Quickly it's confusing to me how a further filter could have gotten into the stopped all state, but it probably has to do with a continuation path. In either case I think more comments would be great.

Reply:
Fixed. Thanks!

This can happen, for example, when a filter calls addDecodedData() in decodeTrailers(), and the next filter in the chain returns stop-all for headers. See testTwoFiltersDecodeHeadersReturnsStopAll() for a case where the condition is needed.